ui/js/services/jobfilters.js

```diff
@@ -331,14 +331,15 @@ treeherder.factory('thJobFilters', [
     };
 
     var toggleInProgress = function() {
-        var rsValues = _toArray($location.search()[QS_RESULT_STATUS]);
-        var pendRun = ['pending', 'running'];
-        if (_.difference(pendRun, rsValues).length === 0) {
-            rsValues = _.without(rsValues, 'pending', 'running');
+        if (_isInProgressShown()) {
```

Comment:
This block can be replaced with this:

    toggleFilters(RESULT_STATUS, ['pending', 'running'], !_isInProgressShown());